------------------------------------------------------------------------
r5690 | cutealien | 2019-01-23 12:49:24 +0000 (Wed, 23 Jan 2019) | 3 lines

Backport from trunk: Fix compilation on OSX and prevent capturing mouse cursor when Window is not on top (Patch #319.
Thanks at Artem Shoobovych for bugreport and patch (https://sourceforge.net/p/irrlicht/patches/319/)


Index: source/Irrlicht/MacOSX/CIrrDeviceMacOSX.mm
===================================================================
--- source/Irrlicht/MacOSX/CIrrDeviceMacOSX.mm	(revision 5689)
+++ source/Irrlicht/MacOSX/CIrrDeviceMacOSX.mm	(revision 5690)
@@ -495,8 +495,8 @@
 		if(!CreationParams.WindowId) //load menus if standalone application
 		{
 			[[NSAutoreleasePool alloc] init];
-			[NSApplication sharedApplication];
-			[NSApp setDelegate:(id<NSFileManagerDelegate>)[[[AppDelegate alloc] initWithDevice:this] autorelease]];
+			[[NSApplication sharedApplication] activateIgnoringOtherApps:YES];
+			[NSApp setDelegate:(id<NSApplicationDelegate>)[[[AppDelegate alloc] initWithDevice:this] autorelease]];
 			[NSBundle loadNibNamed:@"MainMenu" owner:[NSApp delegate]];
 			[NSApp finishLaunching];
 		}

------------------------------------------------------------------------
